google/cloud/security/common/gcp_type/firewall_rule.py

```diff
@@ -278,6 +474,65 @@ def __init__(self, firewall_rules=None, firewall_rule_action='allow'):
 
         self._expanded_rules = None
 
+    def json_dict(self):
+        """Gets the JSON key and values for the firewall action.
+
+        Returns:
+          tuple: Of key ('allowed' or 'denied') and the firewall rules.
+
+        Raises:
+          InvalidFirewallActionError: If a rule is not formatted for the API.
+        """
+        self.validate()
+        if self.action == 'allow':
```

Comment:
Maybe self.action should be 'allowed' and 'denied' instead of 'allow' and 'deny' to match the value in the dict. Then you can just return self.action, self.rules which is a lot cleaner all around.

Reply:
Done.